orderer/common/channelparticipation/restapi.go

```diff
@@ -50,23 +78,122 @@ func (h *HTTPHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	switch req.Method {
-	case http.MethodGet, http.MethodPost, http.MethodDelete:
-		//TODO skeleton
-		err := errors.Errorf("not implemented yet: %s %s", req.Method, req.URL.String())
-		h.sendResponseJsonError(resp, http.StatusNotImplemented, err)
+	h.router.ServeHTTP(resp, req)
+}
+
+// List all channels
+func (h *HTTPHandler) serveListAll(resp http.ResponseWriter, req *http.Request) {
+	_, err := negotiateContentType(req) // Only application/json for now
```

Comment:
routers allows to define matching headers hence you can just define content type there once avoiding repeating this functionality.

Reply:
See above, content negotiation with Accept is not really supported by the mux Headers() matcher.